rotkehlchen/chain/evm/decoding/hop/decoder.py

```diff
@@ -285,6 +295,148 @@ def _decode_token_swap(self, context: DecoderContext) -> DecodingOutput:
+
+        if (swap_asset_id := self.swaps_to_asset.get(context.tx_log.address)) is None:
+            return None
+
+        first_token_amount = self._get_bridge_asset_amount(
+            amount_raw=hex_or_bytes_to_int(context.tx_log.data[96:128]),
+            identifier=swap_asset_id,
+        )
+        second_token_amount = self._get_bridge_asset_amount(
+            amount_raw=hex_or_bytes_to_int(context.tx_log.data[128:160]),
+            identifier=swap_asset_id,
+        )
+        # Filtering out zero amounts to not clash with any other events
+        # And here zero basically means asset not provided for LPing
+        token_amounts = {x for x in (first_token_amount, second_token_amount) if x != ZERO}
+        return user_address, swap_asset_id, token_amounts
+
+    def _handle_event_decoding(
+            self,
+            decoded_events: list[EvmEvent],
+            user_address: str,
+            lp_amount: FVal,
+            token_amounts: list[FVal],
+    ) -> DecodingOutput:
+        """This function is used to enrich the RemoveLiquidity and RemoveLiquidityOne events
+        with proper event types and notes.
+        """
+        for event in decoded_events:
+            if (
+                event.event_type == HistoryEventType.SPEND and
+                event.event_subtype == HistoryEventSubType.NONE and
+                event.location_label == user_address and
+                event.balance.amount == lp_amount
+            ):
+                event.event_subtype = HistoryEventSubType.RETURN_WRAPPED
+                event.counterparty = CPT_HOP
+                event.notes = f'Return {event.balance.amount} {event.asset.symbol_or_name()}'
+            elif (
+                event.event_type == HistoryEventType.RECEIVE and
+                event.event_subtype == HistoryEventSubType.NONE and
+                event.location_label == user_address and
+                event.balance.amount in token_amounts
+            ):
+                event.event_type = HistoryEventType.WITHDRAWAL
+                event.event_subtype = HistoryEventSubType.REMOVE_ASSET
+                event.counterparty = CPT_HOP
+                event.notes = f'Withdraw {event.balance.amount} {event.asset.symbol_or_name()} from Hop'  # noqa: E501
+
+        return DEFAULT_DECODING_OUTPUT
+
+    def _decode_saddle_swap(self, context: DecoderContext) -> DecodingOutput:
+        """This function is used to decode the lp events done via Hop protocol."""
+        if context.tx_log.topics[0] == TOKEN_SWAP:
+            return self._decode_token_swap(context=context)
+
+        if context.tx_log.topics[0] == ADD_LIQUIDITY:
+            return self._decode_add_liquidity(context=context)
+
+        if context.tx_log.topics[0] == REMOVE_LIQUIDITY:
+            return self._decode_remove_liquidity(context=context)
+
+        if context.tx_log.topics[0] == REMOVE_LIQUIDITY_ONE:
+            return self._decode_remove_liquidity_one(context=context)
```

Comment:
unless I'm missing something `_decode_remove_liquidity` and `_decode_remove_liquidity_one` seem to be the same function changing where the LP amount is found. It can be an argument that you can provide here and have one function instead of two